scripts/APIv3.rb

```diff
@@ -2,14 +2,22 @@
 # frozen_string_literal: true
 
 require 'json'
+require 'yaml'
 
 all = {}
 tfa = {}
 regions = {}
 Dir.glob('entries/*/*.json') { |file| all[JSON.parse(File.read(file)).keys[0]] = JSON.parse(File.read(file)).values[0] }
 all.sort.to_h.each do |k, v|
   v['tfa']&.each { |method| (tfa[method].nil? ? tfa[method] = { k => v } : tfa[method][k] = v) }
-  v['regions']&.each { |region| regions[region] = 1 + regions[region].to_i }
+  v['regions']&.each do |region|
+    regions[region] = {} unless regions.key? region
+    regions[region]['count'] = 1 + regions[region]['count'].to_i
+  end
 end
-{ 'all' => all }.merge(tfa).each { |k, v| File.open("api/v3/#{k}.json", 'w') { |file| file.write v.sort_by{ |k,v| k.downcase }.to_json } }
-File.open('api/v3/regions.json', 'w') { |file| file.write regions.sort_by(&:last).reverse.to_h.to_json }
+
+avail_regions = YAML.load_file('_data/regions.yml').group_by { |hash| hash['id'] }.keys
+regions.each { |k, v| v['selection'] = avail_regions.include? k }
+{ 'all' => all }.merge(tfa).each { |k, v| File.open("api/v3/#{k}.json", 'w') { |file| file.write v.sort_by { |k, _v| k.downcase }.to_json } }
+
```

Comment:
```suggestion

regions['int'] = { "count" => all.length, "selection" => true }

```

Adding the `int` region to the `regions.json` file will allow the Cloudflare worker to be simplified by a fair amount:
```js
const base = "https://2fa.directory/";
const status = 302;

async function gatherResponse(response) {
  const { headers } = response;
  const contentType = headers.get("content-type") || "";
  if (contentType.includes("application/json")) {
    return await response.json();
  }
}

async function redirect(request) {
  // Use the cf object to obtain the country of the request.
  // The cf object is only avaialble in production.
  const init = {
    headers: {
      "content-type": "application/json;charset=UTF-8",
    },
  }
  const res = await fetch('https://2fa.directory/api/v3/regions.json', init);
  let countries = await gatherResponse(res);
  const { pathname, search, hash } = new URL(request.url);

  let country = request.cf?.country.toLowerCase() || "int";

  if (!(country in countries) || !countries[country]['selection']){
    country = "int";
  }
  
  return Response.redirect(`${base}${country}${pathname}${search}${hash}`, status);
}

addEventListener("fetch", event => {
  event.respondWith(redirect(event.request));
})

```

Reply:
Thanks for the suggestion!
I made the change to https://redirect.twofactorauth.workers.dev/ but obviously it doesn't work as expected until the regions.json changes are live :wink: